incubator/hnc/pkg/testutils/testutils.go

```diff
@@ -210,22 +210,35 @@ func RunCommand(cmdln ...string) (string, error) {
 	return string(stdout), err
 }
 
+// CleanupNamespaces does everything it can to delete the passed-in namespaces. It also uses very
+// high timeouts (30s) since this function is often called after HNC has just been reinstalled, and
+// it can take a while of HNC to start allowing changes to namespaces again.
 func CleanupNamespaces(nses ...string) {
-	// Remove all possible objections HNC might have to deleting a namesplace. Make sure it
-	// has cascading deletion so we can delete any of its subnamespace descendants, and
-	// make sure that it's not a subnamespace itself so we can delete it directly.
+	// Clean up each namespace
 	for _, ns := range nses {
-		if err := TryRunQuietly("kubectl get ns", ns); err == nil {
-			MustRunWithTimeout(30, "kubectl hns set", ns, "-a")
-			// 'subnamespaceOf' is the old subnamespace annotation used in v0.5. We
-			// still need to clean up this old annotation because this util func is
-			// also used in the API conversion test to clean up namespaces in v0.5.
-			// TODO: remove this line after v0.6 branches and we are no longer
-			//  supporting v1alpha1 conversion.
-			MustRunWithTimeout(30, "kubectl annotate ns", ns, "hnc.x-k8s.io/subnamespaceOf-")
-			MustRunWithTimeout(30, "kubectl annotate ns", ns, "hnc.x-k8s.io/subnamespace-of-")
-			MustRunWithTimeout(30, "kubectl delete ns", ns)
+		// We only check once to see if a namespace exists, but reads are pretty reliable.
+		if err := TryRunQuietly("kubectl get ns", ns); err != nil {
+			continue
 		}
+
+		// Remove all possible objections HNC might have to deleting this namespace. If the HCs are
+		// present, turn on cascading deletion so that we can delete this namespace even if it has
+		// subnamespaces. Note that if HNC isn't properly installed or is installed at an older
+		// version in the conversion tests, this function will fail as it tried to update the HC at a
+		// specific version, so we ignore errors. TODO: find a more reliable way of doing this.
+		TryRun("kubectl hns set", ns, "-a")
```

Comment:
If we are removing namespaces one by one and we will remove annotation for each one, can we just remove `k hns set ns -a`? So `CleanupNamespaces` func can be version neutral.

Reply:
Yeah, good idea. Will do.

Reply:
Done